src/engine/loopingcontrol.cpp

```diff
@@ -910,8 +913,7 @@ void LoopingControl::slotBeatLoop(double beats, bool keepStartPoint, bool enable
     }
 
     int samples = m_pTrackSamples->get();
-    if (!m_pTrack || samples == 0
-            || !m_pBeats) {
+    if ((!m_pTrack || samples == 0 || !m_pBeats) && adjustBeatloopSizeCO) {
         clearActiveBeatLoop();
```

Comment:
Is it correct to skip this as well?

Reply:
Good point, no it is not. I'll just make the following line conditional for adjustBeatloopSizeCO.